### DISCUSSION TITLE
Replace firewall/WFP-based block by "Network adapter not found" or "There is no network"

### DISCUSSION THREAD

**Discussion by ghost** [sandboxie-plus/Sandboxie#3701] [Various ideas and suggestions]
Some software makes repeated attemps to use the Internet to a point of causing launching delays if that software is blocked by firewall. For example MPC-HC + madVR combination takes up to 10 seconds to launch large video files if MPC-HC is blocked with firewall rules. The same delay happens when MPC-HC is launched in Sandboxie sandbox that disallows Internet access. That delay is not present (regardless of sandboxing) if network is physically disconnected. I assume that Sandboxie Internet block is based on WFP because of the mentioned delay. It would be faster if Sandboxie blocked Internet access by telling sandboxed software that there is simply no network connection of any kind or that there is no network adapter. 

Some applicaitons also produce error messages if blocked by firewall, but they don't do that if network is physically disconnected.

**Comment by bastik-1001** [comment]
There are two ways Sandboxie blocks access to the network. The old approach to deny access to devices, which made some apps unhappy and the WFP based filter. There is a per sandbox setting to control what method is used.

In Sandman this can be found in the "Network options" where you can select "Allow access" or "Block using WFP" or "Block by denying access to network devices" from a drop-down menu.

**Comment by ghost** [comment]
Doesn't appear to make any difference to the described delay. Connection is still detected on some level.

**Reply by bastik-1001** [reply]
It was worth a shot, maybe David can look into that and add a blocking method that does not have that delay or modify the existing methods.
